### PR TITLE
LandDA feature.

### PR DESCRIPTION
**Pull request by aerorahul**
Changes need for LandDA PR in global-workflow https://github.com/NOAA-EMC/global-workflow/pull/1635
There is an upstream dependency in NOAA-PSL https://github.com/NOAA-PSL/land-apply_jedi_incr/pull/3

Adds a jinja2 template for a utility used to create analysis from increments.  

The source code is in land-apply_jedi_incr, the template is in GDASApp and it is called in global-workflow.
What could possibly go wrong?

### CODE REVIEW

**Review by jiaruidong2017 — CHANGES_REQUESTED** [review]
Please use the code: `/scratch2/NCEPDEV/stmp1/Jiarui.Dong/gdasapp/gdasapp.9/test/land/apply_jedi_incr.sh` to replace the code `test/land/apply_jedi_incr.sh` in your branch `feature/land_analysis` to avoid the failure of ctest due to the change of source code in land-apply_jedi_incr.

**Comment by aerorahul** [comment]
> Please use the code: `/scratch2/NCEPDEV/stmp1/Jiarui.Dong/gdasapp/gdasapp.9/test/land/apply_jedi_incr.sh` to replace the code `test/land/apply_jedi_incr.sh` in your branch `feature/land_analysis` to avoid the failure of ctest due to the change of source code in land-apply_jedi_incr.

done.

**Review by jiaruidong2017 — DISMISSED** [review]

**Review by jiaruidong2017 — DISMISSED** [review]
Don't forget to change back to PSL's land-apply_jedi_incr.git after it is merged into develop.

**Comment by aerorahul** [comment]
> Don't forget to change back to PSL's land-apply_jedi_incr.git after it is merged into develop.

Yes. Will do. 
@CoryMartin-NOAA 
Please don't merge this till the PSL PR is merged.

I don't know when that will be.

**Comment by jiaruidong2017** [comment]
Please make the following changes for the G-W: 
In `parm/land/prep/prep_ims.yaml`

Change from:
```
ims2ioda:
  copy:
    - [$(DATA)/ims_snow_{{ current_cycle | to_YMDH }}.nc4, $(COM_OBS)/$(OPREFIX)ims_snow.nc4]
```
TO: 
```
ims2ioda:
  copy:
    - [$(DATA)/ims_snow_{{ current_cycle | to_YMDH }}.nc4, $(COM_OBS)/$(OPREFIX)ims_snow_{{ current_cycle | to_YMDH }}.nc4]
```

**Comment by CoryMartin-NOAA** [comment]
@jiaruidong2017 I thought we discussed Tuesday that we were going to remove the cycle time from the obs file?

**Comment by jiaruidong2017** [comment]
> @jiaruidong2017 I thought we discussed Tuesday that we were going to remove the cycle time from the obs file?

@CoryMartin-NOAA My bad, and I deleted my comments above.

**Comment by jiaruidong2017** [comment]
Please copy the obs yaml files: 
```
/scratch1/NCEPDEV/global/Jiarui.Dong/JEDI/GlobalWorkflow/global-workflow.landid8888/sorc/gdas.cd/parm/land/obs/config/adpsfc_snow.yaml
/scratch1/NCEPDEV/global/Jiarui.Dong/JEDI/GlobalWorkflow/global-workflow.landid8888/sorc/gdas.cd/parm/land/obs/config/ims_snow.yaml
```
The above two yaml files are used to replace the obs yaml files at the `parm/land/obs/config/` directory in your branch `feature/land_analysis`.  The above changes just remove the cycle time from the obs file names, and this update will lead to the successful runs for g-w snow DA at all cycles.

**Comment by aerorahul** [comment]
@jiaruidong2017 @CoryMartin-NOAA 
The NOAA-PSL/land-apply_jedi_incr PR https://github.com/NOAA-PSL/land-apply_jedi_incr/pull/3 has been merged.

global-workflow branch has been updated.

This PR is ready for a final review, test and merge.

**Review by jiaruidong2017 — APPROVED** [review]

File: parm/land/obs/config/adpsfc_snow.yaml
Comment (CoryMartin-NOAA) [inline parm/land/obs/config/adpsfc_snow.yaml:13]: Just a note, the current operational convention is to have PDYcyc in the diag filename. Is that something we want to change or continue with? For diags, I would like to keep it if possible. @aerorahul , thoughts?

Comment (aerorahul) [reply parm/land/obs/config/adpsfc_snow.yaml:13]: I just copied the files that @jiaruidong2017 pointed me to.
If I revert them here, an update may be needed in the global-workflow.
What do you want me to do in the light that we will move away from having a time-stamp in this file?

Comment (CoryMartin-NOAA) [reply parm/land/obs/config/adpsfc_snow.yaml:13]: As long as these names are consistent currently between YAML and scripts, let's go with that for now. I would like a more broad discussion with other developers/users before I commit one way or another on the diag files losing dates in their filenames

Comment (jiaruidong2017) [reply parm/land/obs/config/adpsfc_snow.yaml:13]: @aerorahul you can just revert them back and we don't need to make any changes to the global-workflow, because the diagnostic files `diag*.nc4` in `$(DATA)/diags/` will be tarred and gzipped into COM for archive.

**Pull request merged**
